### ISSUE TITLE
Put DestructiveOperations logic into a guice component or merge it into an existing component

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#4665]
The `DestructiveOperations` class is a statefull helper, but initialized several times in the transport actions that rely on its logic. This should become a guice component or merge into an existing component, so it will initialized only once. The same applies for `AutoCreateIndex` class.

Perhaps both classes can be merged into a single component.

**Comment by javanna** [comment]
We briefly discussed this, seems like a valuable improvement, made it an adoptme and low hanging fruit.

**Comment by jimhooker2002** [comment]
This looks like the fix to me: https://github.com/jimhooker2002/elasticsearch/compare/master...jimhooker2002:issue-4665

I'll rebase and submit a pull request unless anyone has any comments before I do that.

**Comment by jimhooker2002** [comment]
I've pinged https://github.com/s1monw over email (original submitter of the code) to take a quick look at my diff and help ensure some level of sanity.  I'm a little bit worried about the comment I removed, specifically the part about repeated log statements.  Hopefully Simon can verify I didn't just introduce a regression.

**Comment by jimhooker2002** [comment]
OK, I've rebased to my repos master branch and this is now part of an existing pull request.

https://github.com/elastic/elasticsearch/pull/12968
